tests/layer_tests/tensorflow_tests/test_tf_AddN.py

```diff
@@ -66,3 +64,51 @@ def test_addn_placeholder_const(self, params, ie_device, precision, ir_version,
                                                           use_legacy_frontend=use_legacy_frontend),
                    ie_device, precision, ir_version, temp_dir=temp_dir,
                    use_legacy_frontend=use_legacy_frontend)
+
+class TestComplexAddN(CommonTFLayerTest):
+    def _prepare_input(self, inputs_info):
+        rng = np.random.default_rng()
+        assert 'param_imag_1:0' in inputs_info
+        assert 'param_real_1:0' in inputs_info
+        assert 'param_real_2:0' in inputs_info
+        assert 'param_imag_2:0' in inputs_info
+        param_real_shape_1 = inputs_info['param_real_1:0']
+        param_imag_shape_1 = inputs_info['param_imag_1:0']
+        param_real_shape_2 = inputs_info['param_real_2:0']
+        param_imag_shape_2 = inputs_info['param_imag_2:0']
+        inputs_data = {}
+        inputs_data['param_real_1:0'] = 4 * rng.random(param_real_shape_1).astype(np.float32) - 2
+        inputs_data['param_imag_1:0'] = 4 * rng.random(param_imag_shape_1).astype(np.float32) - 2
+        inputs_data['param_real_2:0'] = 4 * rng.random(param_real_shape_2).astype(np.float32) - 2
+        inputs_data['param_imag_2:0'] = 4 * rng.random(param_imag_shape_2).astype(np.float32) - 2
+        return inputs_data
+
+    def create_complex_addn_net(self, input_shapes):
+        tf.compat.v1.reset_default_graph()
+        with tf.compat.v1.Session() as sess:
+            param_real_1 = tf.compat.v1.placeholder(np.float32, input_shapes[0], 'param_real_1')
+            param_imag_1 = tf.compat.v1.placeholder(np.float32, input_shapes[1], 'param_imag_1')
+            param_real_2 = tf.compat.v1.placeholder(np.float32, input_shapes[2], 'param_real_2')
+            param_imag_2 = tf.compat.v1.placeholder(np.float32, input_shapes[3], 'param_imag_2')
+            complex_1 = tf.raw_ops.Complex(real=param_real_1, imag=param_imag_1)
+            complex_2 = tf.raw_ops.Complex(real=param_real_2, imag=param_imag_2)
+            addn = tf.raw_ops.AddN(inputs=[complex_1, complex_2], name='complex_AddN')
+            real = tf.raw_ops.Real(input=addn)
+            imag = tf.raw_ops.Imag(input=addn)
+            tf.compat.v1.global_variables_initializer()
+            tf_net = sess.graph_def
+        return tf_net, None
+
+    test_data = [
+        dict(input_shapes=[[1], [1], [1], [1]]),
```

Comment:
please add test cases with one input to `AddN` (i.e. two input_shapes), two (four input_shapes) and three (6 input_shapes) and use for loop for it.

Reply:
Of course! I was unsure whether to do it or not, I'll implement it right away!

Reply:
@rkazants I’ve implemented it.